src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.Repository/HomeRepository.cs

```diff
@@ -52,5 +53,18 @@ public int GetCorrectAnswersCountForUser(string userId)
                 .GroupBy(t => t.TestId)
                 .Count();
         }
+
+        public List<Result> GetAllTestResults()
+        {
+            return _dbContext.Set<Result>()
+                .OrderByDescending(r => r.Points)
+                .ToList();
+        }
+
+        public int GetUserPosition(string userId)
+        {
+            return _dbContext.Set<Result>()
+                .FirstOrDefault(r => r.UserId == userId)?.Id ?? 0;
```

Comment:
it is better to use Index, ex. https://www.codeproject.com/Questions/236792/how-to-get-index-of-list-item-in-Csharp